### PR TITLE
Update to cynic 2.0

### PR DESCRIPTION
**Pull request by obmarg**
I'm the author of cynic, the graphql client library this project is using.  I just released v2 of cynic, so thought I'd raise a PR that updates triagebot to use it.

I've put some effort into improving compile times for large schemas in this release.  In my (not very scientific) tests the `github-graphql` crate in this repo compiles in about half the time after these changes.  Frankly it's still not as fast as I'd like but the github schema is a real challenge.

I also simplified a few of your queries & the associated handling code, hope that's ok.  Happy to revert that stuff if not though.

### CODE REVIEW

**Comment by apiraino** [comment]
hey @obmarg great to see you around :) I love cynic, thanks for this patch!

At a cursory look it looks great. I'd leave it sink for a a bit so I can try it this week for the `T-compiler` meeting.

(but tbh I'm also fine to straight merge this if the triagebot maintainers say so)

**Comment by Mark-Simulacrum** [comment]
@obmarg Hi! Thanks for this patch, this is great. It looks like we've edited the graphql we use a little since you made it (sorry!), but I'd be happy to merge a rebased version of it.

**Comment by obmarg** [comment]
No worries @Mark-Simulacrum - have rebased & updated the new query

**Review by Mark-Simulacrum — APPROVED** [review]

**Pull request merged**
